usaspending_api/search/v2/views/search.py

```diff
@@ -735,6 +737,70 @@ def post(self, request):
         return Response({"results": results})
 
 
+class SpendingByTransactionVisualizationViewSet(APIView):
+
+    @total_ordering
+    class MinType(object):
+        def __le__(self, other):
+            return True
+
+        def __eq__(self, other):
+            return self is other
+    Min = MinType()
+
+    @cache_response()
+    def post(self, request):
+        """Return all budget function/subfunction 
+        titles matching the provided search text"""
+        json_request = request.data
+        fields = json_request.get("fields", None)
+        filters = json_request.get("filters", None)
+        order = json_request.get("order", "asc")
+        limit = json_request.get("limit", 10)
+        page = json_request.get("page", 1)
+        
+        lower_limit = (page - 1) * limit
+        upper_limit = page * limit
+        transaction_types = ["Contracts", "Grants", "Direct Payments", "Loans", "Other"]
+        if fields is None:
+            raise InvalidParameterException("Missing one or more required request parameters: fields")
+        elif len(fields) == 0:
+            raise InvalidParameterException("Please provide a field in the fields request parameter.")
+        if filters is None:
+            raise InvalidParameterException("Missing one or more required request parameters: filters")
+        if "award_type_codes" not in filters:
+            raise InvalidParameterException(
+                "Missing one or more required request parameters: filters['award_type_codes']")
+    
+        if order not in ["asc", "desc"]:
+            raise InvalidParameterException("Invalid value for order: {}".format(order))
+        sort = json_request.get("sort", fields[0])
+        if sort not in fields:
+            raise InvalidParameterException("Sort value not found in fields: {}".format(sort))
+        
+        #fields.append('awarding_agency_id')
+
+        queryset, total = search_transactions(filters, fields, sort, 
```

Comment:
It might be better to not call this a queryset. mostly due to the fact that all other querysets are Django Querysets